README
Task::Calendar

The README  is  used  to  introduce the module and provide instructions on how to
install the module, any machine dependencies it may have (for example C compilers
and installed libraries) and any other information that should be provided before
the module is installed.

A  README  file  is required for CPAN modules since CPAN extracts the README file
from a module distribution so that people browsing the archive can use it to  get
an  idea  of  the  module's  uses.  It  is usually a good idea to provide version
information here so that people can decide whether fixes for the module are worth
downloading.

INSTALLATION

To install this module, run the following commands:

	perl Makefile.PL
	make
	make test
	make install

SUPPORT AND DOCUMENTATION

After installing,  you  can  find  documentation for this module with the perldoc
command.

    perldoc Task::Calendar

You can also look for information at:

    RT, CPAN's request tracker (report bugs here)
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Task-Calendar

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/Task-Calendar

    CPAN Ratings
        http://cpanratings.perl.org/d/Task-Calendar

    Search CPAN
        http://search.cpan.org/dist/Task-Calendar/

LICENSE AND COPYRIGHT

Copyright (C) 2015 Mohammad S Anwar

This program is free software; you can redistribute it and/or modify it under the
terms of the the Artistic License (2.0).You may obtain a copy of the full license
at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying / distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you, you're nevertheless required to ensure that your Modified Version
complies with the requirements of this license.

This  license  does  not  grant you the right to use any trademark, service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to  make,  have made, use, offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by the  Package.  If you institute patent litigation
(including  a  cross-claim or  counterclaim)  against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF     MERCHANTABILITY,   FITNESS   FOR   A PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL, OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE,EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

Changes
Revision history for Task::Calendar

0.42  2019-07-25
      - Updated dependency on Calendar::Bahai v0.53.
      - Updated dependency on Calendar::Gregorian v0.22.
      - Updated dependency on Calendar::Hebrew v0.12.
      - Updated dependency on Calendar::Hijri v0.40.
      - Updated dependency on Calendar::Julian v0.13.
      - Updated dependency on Calendar::Persian v0.42.
      - Updated dependency on Calendar::Saka v1.41.

0.41  2019-06-12
      - Updated dependency on Calendar::Bahai v0.52.
      - Updated dependency on Calendar::Gregorian v0.21.
      - Updated dependency on Calendar::Hebrew v0.11.
      - Updated dependency on Calendar::Hijri v0.39.
      - Updated dependency on Calendar::Julian v0.12.
      - Updated dependency on Calendar::Persian v0.41.
      - Updated dependency on Calendar::Saka v1.40.

0.40  2019-05-26
      - Updated dependency on Calendar::Bahai v0.51.
      - Updated dependency on Calendar::Gregorian v0.20.
      - Updated dependency on Calendar::Hebrew v0.10.
      - Updated dependency on Calendar::Hijri v0.38.
      - Updated dependency on Calendar::Julian v0.11.
      - Updated dependency on Calendar::Persian v0.40.
      - Updated dependency on Calendar::Saka v1.39.

0.39  2019-04-10
      - Updated dependency on Calendar::Bahai v0.50.
      - Updated dependency on Calendar::Gregorian v0.19.
      - Updated dependency on Calendar::Hebrew v0.09.
      - Updated dependency on Calendar::Hijri v0.37.
      - Updated dependency on Calendar::Julian v0.10.
      - Updated dependency on Calendar::Persian v0.39.
      - Updated dependency on Calendar::Saka v1.38.

0.38  2019-03-30
      - Updated dependency on Calendar::Bahai v0.49.
      - Updated dependency on Calendar::Gregorian v0.18.
      - Updated dependency on Calendar::Hebrew v0.08.
      - Updated dependency on Calendar::Hijri v0.36.
      - Updated dependency on Calendar::Julian v0.09.
      - Updated dependency on Calendar::Persian v0.38.
      - Updated dependency on Calendar::Saka v1.37.

0.37  2018-12-19
      - Updated dependency on Calendar::Hebrew v0.07.
      - Updated dependency on Calendar::Julian v0.08.

0.36  2018-12-04
      - Updated dependency on Calendar::Bahai v0.48.
      - Updated dependency on Calendar::Gregorian v0.17.
      - Updated dependency on Calendar::Hebrew v0.06.
      - Updated dependency on Calendar::Hijri v0.35.
      - Updated dependency on Calendar::Julian v0.07.
      - Updated dependency on Calendar::Persian v0.37.
      - Updated dependency on Calendar::Saka v1.36.

0.35  2018-05-15
      - Updated dependency on Calendar::Gregorian v0.16.
      - Updated dependency on Calendar::Hebrew v0.05.

0.34  2018-01-30
      - Tidied up Changes file as per CPAN::Changes::Spec.
      - Added test script t/changes.t.

0.33  2018-01-29
      - Upgraded to use Calendar::Julian v0.07.

0.32  2017-09-10
      - Upgraded to use Calendar::Hebrew v0.02.

0.31  2017-07-24
      - Upgraded to use Calendar::Hebrew v0.01.
      - Upgraded to use Calendar::Julian v0.02.

0.30  2017-02-02
      - Upgraded to use Calendar::Bahai v0.46.
      - Upgraded to use Calendar::Gregorian v0.15.
      - Upgraded to use Calendar::Hijri v0.33.
      - Upgraded to use Calendar::Persian v0.35.
      - Upgraded to use Calendar::Saka v1.34.

0.29  2017-01-12
      - Upgraded to use Calendar::Bahai v0.45.
      - Upgraded to use Calendar::Gregorian v0.14.
      - Upgraded to use Calendar::Hijri v0.32.
      - Upgraded to use Calendar::Persian v0.34.
      - Upgraded to use Calendar::Saka v1.33.

0.28  2017-01-05
      - Upgraded to use Calendar::Bahai v0.44.
      - Upgraded to use Calendar::Gregorian v0.13.
      - Upgraded to use Calendar::Hijri v0.31.
      - Upgraded to use Calendar::Persian v0.33.
      - Upgraded to use Calendar::Saka v1.32.

0.27  2016-12-24
      - Upgraded to use Calendar::Bahai v0.43.
      - Upgraded to use Calendar::Gregorian v0.12.
      - Upgraded to use Calendar::Hijri v0.30.
      - Upgraded to use Calendar::Persian v0.32.

0.26  2016-12-12
      - Upgraded to use Calendar::Bahai v0.42.
      - Upgraded to use Calendar::Gregorian v0.11.
      - Upgraded to use Calendar::Hijri v0.29.
      - Upgraded to use Calendar::Persian v0.31.
      - Upgraded to use Calendar::Saka v1.31.

0.25  2016-11-27
      - Upgraded to use Calendar::Bahai v0.41.

0.24  2016-11-20
      - Upgraded to use Calendar::Bahai v0.39.
      - Upgraded to use Calendar::Gregorian v0.10.
      - Upgraded to use Calendar::Hijri v0.28.
      - Upgraded to use Calendar::Persian v0.30.
      - Upgraded to use Calendar::Saka v1.30.

0.23  2016-10-21
      - Upgraded to use Calendar::Bahai v0.38.
      - Upgraded to use Calendar::Gregorian v0.09.
      - Upgraded to use Calendar::Hijri v0.27.
      - Upgraded to use Calendar::Persian v0.29.
      - Upgraded to use Calendar::Saka v1.29.

0.22  2016-09-30
      - Upgraded to use Calendar::Gregorian v0.08
      - Upgraded to use Calendar::Saka v1.28
      - Upgraded to use Calendar::Bahai v0.37

0.21  2016-09-16
      - Upgraded to use Calendar::Hijri v0.26
      - Upgraded to use Calendar::Persian v0.28.

0.20  2016-08-21
      - Upgraded to use Calendar::Bahai v0.36.
      - Upgraded to use Calendar::Gregorian v0.07.
      - Upgraded to use Calendar::Hijri v0.25.
      - Upgraded to use Calendar::Persian v0.27.
      - Upgraded to use Calendar::Saka v1.27.

0.19  2016-07-14
      - Upgraded to use Calendar::Bahai v0.35.
      - Upgraded to use Calendar::Gregorian v0.06.
      - Upgraded to use Calendar::Hijri v0.24.
      - Upgraded to use Calendar::Persian v0.26.
      - Upgraded to use Calendar::Saka v1.26.
      - Updated copyright year information in README.

0.18  2016-05-29
      - Upgraded to use Calendar::Bahai v0.34.
      - Upgraded to use Calendar::Gregorian v0.05.
      - Upgraded to use Calendar::Hijri v0.23.
      - Upgraded to use Calendar::Persian v0.25.
      - Upgraded to use Calendar::Saka v1.25.
      - Tidied up unit test in general.


lib/Task/Calendar.pm
package Task::Calendar;

$Task::Calendar::VERSION   = '0.42';
$Task::Calendar::AUTHORITY = 'cpan:MANWAR';

use 5.006;
use strict; use warnings;

=head1 NAME

Task::Calendar - Bundles Calendar::* packages.

=head1 VERSION

Version 0.42

=head1 DESCRIPTION

It complements package L<App::calendr> and installs  all the supported calendars.

=head1 SYNOPSIS

    cpanm Task::Calendar

=head1 AUTHOR

Mohammad S Anwar, C<< <mohammad.anwar at yahoo.com> >>

=head1 REPOSITORY

L<https://github.com/manwar/Task-Calendar>

=head1 SEE ALSO

=over 4

=item L<Calendar::Bahai>

=item L<Calendar::Gregorian>

=item L<Calendar::Hebrew>

=item L<Calendar::Hijri>

=item L<Calendar::Julian>

=item L<Calendar::Persian>

=item L<Calendar::Saka>

=back

=head1 BUGS

Please report any bugs or feature requests to C<bug-task-calendar at rt.cpan.org>,
or through the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Task-Calendar>.
I will  be notified and then you'll automatically be notified of progress on your
bug as I make changes.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Task::Calendar

You can also look for information at:

=over 4

=item * RT: CPAN's request tracker (report bugs here)

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Task-Calendar>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Task-Calendar>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Task-Calendar>

=item * Search CPAN

L<http://search.cpan.org/dist/Task-Calendar/>

=back

=head1 LICENSE AND COPYRIGHT

Copyright (C) 2015 - 2016 Mohammad S Anwar.

This program  is  free software; you can redistribute it and / or modify it under
the  terms  of the the Artistic License (2.0). You may obtain  a copy of the full
license at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying or distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you,you are nevertheless required to ensure that your Modified Version
 complies with the requirements of this license.

This  license  does  not grant you the right to use any trademark,  service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to make,  have made, use,  offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by  the  Package. If you institute patent litigation
(including  a  cross-claim  or  counterclaim) against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF   MERCHANTABILITY,   FITNESS   FOR   A   PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL,  OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

=cut

1; # End of Task::Calendar


Makefile.PL
#!/usr/bin/perl

use 5.006;
use strict;
use warnings FATAL => 'all';
use ExtUtils::MakeMaker;

WriteMakefile(
    NAME               => 'Task::Calendar',
    AUTHOR             => q{Mohammad S Anwar <mohammad.anwar@yahoo.com>},
    VERSION_FROM       => 'lib/Task/Calendar.pm',
    ABSTRACT_FROM      => 'lib/Task/Calendar.pm',
    LICENSE            => 'artistic_2',
    MIN_PERL_VERSION   => 5.006,
    CONFIGURE_REQUIRES => {
        'ExtUtils::MakeMaker' => 0,
    },
    BUILD_REQUIRES   => {
        'Test::More' => 0,
    },
    PREREQ_PM => {
        'Calendar::Bahai'     => '0.45',
        'Calendar::Gregorian' => '0.14',
        'Calendar::Hijri'     => '0.32',
        'Calendar::Persian'   => '0.34',
        'Calendar::Saka'      => '1.33',
    },
    dist  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean => { FILES => 'Task-Calendar-*' },
    (eval { ExtUtils::MakeMaker->VERSION(6.46) } ? (META_MERGE => {
        'meta-spec' => { version => 2 },
        resources => {
            repository => {
                type => 'git',
                url  => 'https://github.com/manwar/Task-Calendar.git',
                web  => 'https://github.com/manwar/Task-Calendar',
            },
        }})
     : ()
    ),
);
